cxx/util_distribution_variant.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024
+// See LICENSE.txt
+
+#include "util_distribution_variant.hh"
+
+#include <cassert>
+#include <sstream>
+
+#include "distributions/beta_bernoulli.hh"
+#include "distributions/bigram.hh"
+#include "distributions/crp.hh"
+#include "distributions/dirichlet_categorical.hh"
+#include "distributions/normal.hh"
+#include "domain.hh"
+#include "relation.hh"
+
+ObservationVariant observation_string_to_value(
+    const std::string& value_str, const DistributionEnum& distribution) {
+  switch (distribution) {
+    case DistributionEnum::normal:
+      return std::stod(value_str);
+    case DistributionEnum::bernoulli:
+      return std::stod(value_str);
+    case DistributionEnum::categorical:
+      return std::stoi(value_str);
+    case DistributionEnum::bigram:
+      return value_str;
+    default:
+      assert(false);
```

Comment:
Here and below, printing an error message before aborting would be nice.

Reply:
Done.